tpu-provisioner/internal/cloud/gke.go

```diff
@@ -33,6 +33,7 @@ const (
 	GoogleTPUResource          = "google.com/tpu"
 	gcpLabelPrefix             = "cloud.google.com/"
 	googleLabelPrefix          = "google.com/"
+	ICIResiliencyLabel         = "cloud.google.com/gke-tpu-ici-resiliency"
```

Comment:
Does the label control ici-resiliency or is the label applied based on an option supplied when creating the node pool?

Reply:
The latter, we are using the labels from node selector to instruct provisioner or NAP to create the node pools. GKE control plane will interpret this info and pass the intent via VM metadata to GCE.